tools/gen/impl_template

```diff
@@ -321,8 +321,10 @@ int ${tlv.name}_is_valid(const struct ${tlv.struct_name()} *record)
 			 *   - otherwise, if `type` is odd:
 			 *     - MUST discard the next `length` bytes.
 			 */
-			SUPERVERBOSE("Unknown even type in TLV");
-			return i;
+			SUPERVERBOSE("unknown even");
+			if (err_index != NULL)
```

Comment:
is this better than `if (err_index)` ?

Reply:
I slightly prefer it since it makes it clearer that we aren't checking a boolean but rather a pointer. I'm pretty sure it results in the same assembly.

Reply:
Indeed: https://godbolt.org/z/tcXEAc

Reply:
personally, i prefer the more concise `if (err_index)`; the fact that `err_index`'s type is a pointer should be clue enough that we're checking for pointer presence